Building on ocaml/ocaml trunk broken after upstream signal changes

## Expected Behavior

dune builds with ocaml/ocaml trunk

## Actual Behavior

dune fails to build with ocaml/ocaml trunk
https://github.com/ocaml-multicore/multicoretests/actions/runs/14566902818/job/40857535173
```
  ocamlc -output-complete-exe -w -24 -g -o .duneboot.exe -I boot -I +unix unix.cma boot/libs.ml boot/duneboot.ml
  ./.duneboot.exe
  cd _boot && /home/runner/local/bin/ocamlopt.opt -c -g -no-alias-deps -w -49-6 -alert -unstable -I +unix -I +threads ocamlc_loc.mli
  File "otherlibs/ocamlc-loc/src/ocamlc_loc.mli", line 1, characters 4-9:
  Warning 53 [misplaced-attribute]: the alert attribute cannot appear in this context
  cd _boot && /home/runner/local/bin/ocamlopt.opt -c -g -no-alias-deps -w -49-6 -alert -unstable -I +unix -I +threads stdune__Signal.ml
  File "otherlibs/stdune/src/signal.ml", line 65, characters 11-19:
  Error: This expression has type int
         This is not a function; it cannot be applied.
  
  make: *** [Makefile:32: _boot/dune.exe] Error 2
  
  Error: Process completed with exit code 2.
```

The problem is that https://github.com/ocaml/ocaml/pull/13843 added a definition for `sigwinch`:
```ocaml
val sigwinch : signal
```
which causes this (previous work-around) line to fail type checking:

https://github.com/ocaml/dune/blob/0764b1a39906bdd771e74d9d862062251a874bfb/otherlibs/stdune/src/signal.ml#L65

## Reproduction

1. install an ocaml/ocaml trunk compiler
2. `make release && make install`

- Version of `dune` (output of `dune --version`): 3.18.0
- Version of `ocaml` (output of `ocamlc --version`): ocaml/ocaml trunk built from commit 765fcdb83a487ddbd4bcdd92cc47823dd79f700b
- Operating system (distribution and version): all (Linux/macOS/Cygwin/MinGW/MSVC)

--- a/otherlibs/stdune/src/signal.ml
+++ b/otherlibs/stdune/src/signal.ml
@@ -33,35 +33,34 @@ type t =
 external sigwinch : unit -> int = "stdune_winch_number" [@@noalloc]
 
 let all =
-  let open Sys in
-  [ Abrt, sigabrt
-  ; Alrm, sigalrm
-  ; Fpe, sigfpe
-  ; Hup, sighup
-  ; Ill, sigill
-  ; Int, sigint
-  ; Kill, sigkill
-  ; Pipe, sigpipe
-  ; Quit, sigquit
-  ; Segv, sigsegv
-  ; Term, sigterm
-  ; Usr1, sigusr1
-  ; Usr2, sigusr2
-  ; Chld, sigchld
-  ; Cont, sigcont
-  ; Stop, sigstop
-  ; Tstp, sigtstp
-  ; Ttin, sigttin
-  ; Ttou, sigttou
-  ; Vtalrm, sigvtalrm
-  ; Prof, sigprof
-  ; Bus, sigbus
-  ; Poll, sigpoll
-  ; Sys, sigsys
-  ; Trap, sigtrap
-  ; Urg, sigurg
-  ; Xcpu, sigxcpu
-  ; Xfsz, sigxfsz
+  [ Abrt, Sys.sigabrt
+  ; Alrm, Sys.sigalrm
+  ; Fpe, Sys.sigfpe
+  ; Hup, Sys.sighup
+  ; Ill, Sys.sigill
+  ; Int, Sys.sigint
+  ; Kill, Sys.sigkill
+  ; Pipe, Sys.sigpipe
+  ; Quit, Sys.sigquit
+  ; Segv, Sys.sigsegv
+  ; Term, Sys.sigterm
+  ; Usr1, Sys.sigusr1
+  ; Usr2, Sys.sigusr2
+  ; Chld, Sys.sigchld
+  ; Cont, Sys.sigcont
+  ; Stop, Sys.sigstop
+  ; Tstp, Sys.sigtstp
+  ; Ttin, Sys.sigttin
+  ; Ttou, Sys.sigttou
+  ; Vtalrm, Sys.sigvtalrm
+  ; Prof, Sys.sigprof
+  ; Bus, Sys.sigbus
+  ; Poll, Sys.sigpoll
+  ; Sys, Sys.sigsys
+  ; Trap, Sys.sigtrap
+  ; Urg, Sys.sigurg
+  ; Xcpu, Sys.sigxcpu
+  ; Xfsz, Sys.sigxfsz
   ; Winch, sigwinch ()
   ]
 ;;

